docs/developers/define-secrets.rst

```diff
@@ -0,0 +1,188 @@
+##############################
+Define the application secrets
+##############################
+
+To allow Phalanx to manage and verify the secrets for an application, every secret that the application uses must be configured.
+This configuration is used to generate secrets that can be randomly generated, copy secret values from an external source if appropriate, and ensure that all secrets required for the application are present in Vault.
+
+If the application doesn't have any secrets, you can skip this step and continue to :doc:`add-application`.
+
+If you have not already written a Helm chart for your application, do that first, following the instructions in either :doc:`write-a-helm-chart` or :doc:`add-external-chart`.
+
+Create the secret definitions
```

Comment:
Although the schema reference is great, it'd still be good to have some examples for different scenarios that someone can look at and copy.

Reply:
Examples are a great idea, thanks! Now added.